frontend/packages/topology/src/bindable-services/data-transformer.ts

```diff
@@ -0,0 +1,95 @@
+import { EdgeModel, Model, NodeModel } from '@patternfly/react-topology';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { TYPE_SERVICE_BINDING } from '@console/topology/src/const';
+import { getTopologyNodeItem } from '@console/topology/src/data-transforms/transform-utils';
+import { edgesFromServiceBinding } from '@console/topology/src/operators/operators-data-transformer';
+import { TopologyDataObject } from '@console/topology/src/topology-types';
+import {
+  fetchBindableServices,
+  getBindableServicesList,
+  getServiceBindingsList,
+} from './fetch-bindable-services-utils';
+
+export const TYPE_BINDABLE_SERVICE_NODE = 'bindable-node';
+
+// Used to perform discovery of bindable services on console load since topology data model loader requires models upfront
+fetchBindableServices();
+
+const getTopologyBindableServiceNodes = (resources: K8sResourceKind[]): NodeModel[] => {
+  const nodes = resources.map((obj) => {
+    const data: TopologyDataObject = {
+      id: obj.metadata.uid,
+      name: obj.metadata.name,
+      type: TYPE_BINDABLE_SERVICE_NODE,
+      resource: obj,
+      resources: { obj },
+      data: {
+        resource: obj,
+      },
+    };
+    return getTopologyNodeItem(obj, TYPE_BINDABLE_SERVICE_NODE, data);
+  });
+
+  return nodes;
+};
+
+const getTopologyBindableServiceEdges = (
+  resource: K8sResourceKind,
+  nodes: NodeModel[],
+  sbrs: K8sResourceKind[],
+): EdgeModel[] => {
+  const edges = [];
+  if (!sbrs?.length || !nodes?.length) {
+    return edges;
+  }
+
+  edgesFromServiceBinding(resource, sbrs).forEach((sbr) => {
+    sbr.spec.services?.forEach((bss) => {
+      if (bss) {
+        const targetNode = nodes.find(
+          (node) =>
+            node.data.resource.kind === bss.kind && node.data.resource.metadata.name === bss.name,
+        );
+        if (targetNode) {
+          const target = targetNode.data.resource.metadata.uid;
+          const source = resource.metadata.uid;
+          if (source && target) {
+            edges.push({
+              id: `${source}_${target}`,
+              type: TYPE_SERVICE_BINDING,
+              source,
+              target,
+              resource: sbr,
+              data: { sbr },
+            });
+          }
+        }
+      }
+    });
+  });
+
+  return edges;
+};
+
+export const getBindableServicesTopologyDataModel = async (namespace: string): Promise<Model> => {
+  const bindableResources = await getBindableServicesList(namespace);
+  const serviceBindings = await getServiceBindingsList(namespace);
```

Comment:
No need to fetch these resources separate,  this function will have two arguments `namespace` and `resources`. resources will contain all the resources fetched through topology.